https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/3593
https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/93d5848031cd9101d830fb4c37c5a7404ac5f276

From 93d5848031cd9101d830fb4c37c5a7404ac5f276 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Sat, 21 Oct 2023 09:27:43 +0200
Subject: [PATCH] module-echo-cancel: playback and source are async

The playback and source streams don't dequeue/queue buffers from
the process function and so need to be marked async.

Fixes #3593
--- a/src/modules/module-echo-cancel.c
+++ b/src/modules/module-echo-cancel.c
@@ -1002,7 +1002,8 @@ static int setup_streams(struct impl *impl)
 			PW_DIRECTION_OUTPUT,
 			PW_ID_ANY,
 			PW_STREAM_FLAG_MAP_BUFFERS |
-			PW_STREAM_FLAG_RT_PROCESS,
+			PW_STREAM_FLAG_RT_PROCESS |
+			PW_STREAM_FLAG_ASYNC,
 			params, n_params)) < 0) {
 		spa_pod_dynamic_builder_clean(&b);
 		return res;
@@ -1036,7 +1037,8 @@ static int setup_streams(struct impl *impl)
 			PW_ID_ANY,
 			PW_STREAM_FLAG_AUTOCONNECT |
 			PW_STREAM_FLAG_MAP_BUFFERS |
-			PW_STREAM_FLAG_RT_PROCESS,
+			PW_STREAM_FLAG_RT_PROCESS |
+			PW_STREAM_FLAG_ASYNC,
 			params, n_params)) < 0) {
 		spa_pod_dynamic_builder_clean(&b);
 		return res;
-- 
GitLab
